tests/basic/pure_test.go

```diff
@@ -5002,3 +5004,136 @@ var _ = Describe("{RebootingNodesWhileFADAvolumeCreationInProgressUsingZones}",
 		AfterEachTest(contexts)
 	})
 })
+
+var _ = Describe("{ValidatePodNameinVolume}", func() {
+	/*
+	   https://purestorage.atlassian.net/browse/PWX-37369
+	   As part of Mulitenancy feature, we are deploying an app which has "pure_fa_pod_name" in the storage class
+	   1. Loop Through pure.json file and pick an FA endpoint which has Realm, if FA is not accessible go through next FA, if none accessible Fail the Test
+	   2. Create a pod inside the Realm which we got from first step
+	   3. Create a volume using the same pod name that is created in the FA ,The pod name should be mentioned as pure_fa_pod_name in the storage class
+	   4. Validate Application and Check if the pod name in the volume is same as the pod name in the storage class
+	   5. Delete the Application and the pod created in the FA (Right now we only destroy the pod , as delete pod will not happen because eradication is blocked by SafeMode in FA)
+	*/
+	JustBeforeEach(func() {
+		StartTorpedoTest("ValidatePodNameinVolume", "Validate the pod name in the volume", nil, 0)
+	})
+	var contexts []*scheduler.Context
+	itLog := "ValidatePodNameinVolume"
+	It(itLog, func() {
+		log.InfoD(itLog)
+		var origCustomAppConfigs map[string]scheduler.AppConfig
+		var RealmName string
+		var faClient *newFlashArray.Client
+		var isFAaccessible bool
+		testName := "validate-pod-name-in-volume"
+
+		flashArrays, err := GetFADetailsUsed()
+		log.FailOnError(err, "Failed to get FA details from pure.json in the cluster")
+		for _, fa := range flashArrays {
+			if fa.Realm != "" {
+				RealmName = fa.Realm
+				faClient, err = pureutils.PureCreateClientAndConnectRest2_x(fa.MgmtEndPoint, fa.APIToken)
+				if err != nil {
+					log.Errorf("Failed to connect to FA using Mgmt IP [%v]", fa.MgmtEndPoint)
+					continue
+				}
+				isFAaccessible = true
+				break
+			}
+		}
+		if !isFAaccessible {
+			log.FailOnError(fmt.Errorf("No FA with realm found in pure.json"), "No FA with realm found in pure.json")
+		}
+		isRealmExists, err := pureutils.IsFARealmExistsOnMgmtEndpoint(faClient, RealmName)
+		log.FailOnError(err, fmt.Sprintf("Failed to check if realm [%v] exists ", RealmName))
+		if !isRealmExists {
+			log.FailOnError(fmt.Errorf("Realm [%v] is not created in FA", RealmName), "is realm created in FA?")
+		}
+		podNameinSC := "Torpedo-Test" + Inst().InstanceID
+		PodNameinFA := RealmName + "::" + podNameinSC
+
+		stepLog := "Create A pod inside Realm"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			_, err = pureutils.CreatePodinFA(faClient, PodNameinFA)
```

Comment:
do you first want to check if the pod already exists before creating?

Reply:
Added it